person.rb

```diff
@@ -1,18 +1,26 @@
+require './corrector'
+
 class Person
   attr_accessor :name, :age
   attr_reader :id
 
   def initialize(age, name = 'Unknown', parent_permission: true)
-    @id = Random.ran(1..1000)
+    @id = Random.rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
+    @corrector = Corrector.new
   end
 
   def of_age?
     @age >= 18
   end
 
+  def validate_name
+    validate = @corrector
+    @name = validate.correct_name(@name)
```

Comment:
[OPTIONAL]

- Instead of assigning to another variable and then calling, you can directly call the correct_name method as well. [ like `@corrector.correct_name(@name)`]

Reply:
Thank you again for explaining to me what to do. I change everything that you asked.